src/map.cpp

```diff
@@ -1085,22 +851,23 @@ void Map::addNodeAndUpdate(v3s16 p, MapNode n,
 	}
 
 	/*
-		Add neighboring liquid nodes and the node itself if it is
-		liquid (=water node was added) to transform queue.
-	*/
+		Add neighboring liquid nodes and this node to transform queue.
+		(it's vital for the node itself to get updated last, if it was removed.)
```

Comment:
To tell the truth, I don't know. Since it was written in `Map::removeNodeAndUpdate`, I didn't want to alter it, because I only wanted to modify code related to lighting and not to introduce liquid flow bugs.
I don't know who wrote this comment originally, but I would be very thankful if he/she explained there why is this important.
If you want me, I will try to find it out myself, but I think it is related to liquids, not lights.